### PR TITLE
Fixes & spec tests

### PR DESCRIPTION
**Pull request by Phil-Friderici**

### CODE REVIEW

**Comment by Phil-Friderici** [comment]
Fixed an dependency issue when $service_manage was set to false.

File: spec/classes/init_spec.rb [hunk 1]
```diff
@@ -191,12 +190,219 @@
...
+    end
+
+    context 'when check_interval is set to valid integer <242>' do
+      let(:params) { { :check_interval => 242 } }
+      it { should contain_file('monit_config').with_content(/^set daemon 242$/) }
+    end
+
+    context 'when httpd is set to valid bool <true>' do
+      let(:params) { { :httpd => true } }
+      content = <<-END.gsub(/^\s+\|/, '')
```
Comment (ghoneycutt) [inline spec/classes/init_spec.rb:219]: This looks magical. What does it do?

**Comment by ghoneycutt** [comment]
Thanks!!

File: spec/classes/init_spec.rb (hunk 1 above)
Comment (Phil-Friderici) [inline spec/classes/init_spec.rb:219]: The .gsub removes the leading spaces and delimiting pipe "|" from the following example. So you can have proper indenting which improves the readability.

Comment (ghoneycutt) [inline spec/classes/init_spec.rb:219]: Very clever! How do you think it compares to using fixtures or putting the strings in a `%{}` such as in https://github.com/ghoneycutt/puppet-module-dnsclient/blob/master/spec/classes/init_spec.rb#L19-23

Comment (Phil-Friderici) [inline spec/classes/init_spec.rb:219]: I don't like that using %{}, you loose proper indenting and therefore it disturbs readability. Which is kept with the help of gsub().

For fixtures I have the same feeling like for params.pp ;)
I like them for a fully feature test to ensure all formating and sequencing details. Afterwards I prefer to only test the changed content, as long as we are talking about less than lets say 5 or 6 lines.

Comment (ghoneycutt) [inline spec/classes/init_spec.rb:219]: Makes sense to me. Thanks for explaining.

Comment (Phil-Friderici) [inline spec/classes/init_spec.rb:219]: my pleasure

**Pull request merged**